tests/cpp/test_cpu_id_hash_map.cc

```diff
@@ -0,0 +1,60 @@
+#include <set>
+#include <gtest/gtest.h>
+#include <dgl/array.h>
+
+#include "../../src/array/cpu/cpu_id_hash_map.h"
+#include "./common.h"
+
+using namespace dgl;
+using namespace dgl::runtime;
+using namespace dgl::aten;
+
+namespace {
+
+template <typename IdType>
+void ConstructRandomSet(size_t size, size_t range,
+  std::vector<IdType>& id_vec) {
+    id_vec.resize(size);
+    std::srand(42);
+    for (size_t i = 0; i < size; i++) {
+        id_vec[i] = static_cast<IdType>(std::rand() % range);
+    }
+}
+
+template <typename IdType, size_t size, size_t range>
+void _TestIdMap() {
+    std::vector<IdType> id_vec;
+    ConstructRandomSet(size, range, id_vec);
+    std::set<IdType> id_set(id_vec.begin(), id_vec.end());
+    IdArray ids = VecToIdArray(id_vec, sizeof(IdType) * 8, CTX);
+    IdArray unique_ids = NewIdArray(size, CTX, sizeof(IdType) * 8);
+    CpuIdHashMap<IdType> id_map(CTX);
+    auto unique_num = id_map.Init(ids, unique_ids);
+    unique_ids -> shape[0] = unique_num;
+    IdType* unique_id_data = unique_ids.Ptr<IdType>();
+    EXPECT_EQ(id_set.size(), unique_num);
+
+    for (size_t i = 0; i < unique_num; i++) {
```

Comment:
it it possible to check in parallel which may benefit in large size map test?

Reply:
sure, use parallel_for for acceleration.